drivers/timer/Kconfig

```diff
@@ -196,6 +196,7 @@ config NATIVE_POSIX_TIMER
 	bool "(POSIX) native_posix timer driver"
 	default y
 	depends on BOARD_NATIVE_POSIX
+	select TICKLESS_CAPABLE
```

Comment:
FWIW: I'd sell this bit in the commit message, because that's what's actually being added here semantically.  The reason this was still using the legacy API is that it was still a simple repeating timer interrupt and I wasn't 100% how to make tickless work using the native_posix environment.

Reply:
done :)